### PR TITLE
update FAQ, add one about slow luarocks

### PR DESCRIPTION
**Pull request by idevz**
# Summary: a FAQ about slow luarocks issue

a method of luarocks usage to avoid slow or timeout issue

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -56,3 +56,15 @@ For the configuration center, configuration storage is only the most basic funct
...
+
+## Why is it that installing APISIX dependencies with Luarocks causes timeout, slow or unsuccessful installation?
+
+There are two possibilities when encountering slow luarocks:
+
+1. Server used for luarocks installation is blocked
+2. There is a place between your network and github server to block the 'git' protocol
+
+For the first problem, you can use https_proxy or use the `--server` option to specify a luarocks server that you can access or access faster.
+Run the `luarocks config rocks_servers` command to see which server are available.
```
Comment (membphis) [inline FAQ.md]: My Luarocks is too old?

```
$ luarocks config rocks_servers

Error: Please provide a flag for querying configuration values. See '/usr/bin/luarocks help config'.

$ luarocks --version
/usr/bin/luarocks 2.3.0
LuaRocks main command-line interface
```

Comment (idevz) [reply FAQ.md]: very very old.

```bash
 ~/ luarocks --version
/usr/local/bin/luarocks 3.2.1
LuaRocks main command-line interface

 ~/ luarocks config rocks_servers
{
   {
      "https://luarocks.org",
      "https://raw.githubusercontent.com/rocks-moonscript-org/moonrocks-mirror/master/",
      "http://luafr.org/moonrocks/",
      "http://luarocks.logiceditor.com/rocks"
   }
}
```

Comment (idevz) [reply FAQ.md]: here is a example about install protobuf. I using a different server `--server='http://luafr.org/moonrocks/'`

```bash
 ~/code/iresty/apisix/ [master*] luarocks --server='http://luafr.org/moonrocks/' --verbose install --lua-dir=${LUA_JIT_DIR} lua-protobuf --tree=deps --only-deps --local

os.execute: 	"/usr/local/Cellar/openresty/1.15.8.2/luajit/bin/luajit" -e "io.write(tostring(jit and jit.version:gsub([[^%w-JIT ]], [[]])))" > /tmp/lua_VkrykN
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

os.execute: 	curl '--version'
curl 7.65.3 (x86_64-apple-darwin13.4.0) libcurl/7.65.3 OpenSSL/1.1.1d zlib/1.2.11 libssh2/1.8.2
Release-Date: 2019-07-19
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS GSS-API HTTPS-proxy IPv6 Kerberos Largefile libz NTLM NTLM_WB SPNEGO SSL TLS-SRP UnixSockets
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

os.execute: 	curl -k -f -L --user-agent "LuaRocks/3.2.1 macosx-x86_64 via curl" --connect-timeout 30 'http://luafr.org/moonrocks/manifest-5.1.zip' > '/Users/idevz/.cache/luarocks/http___luafr.org_moonrocks_/manifest-5.1.zip' 2> /dev/null
Results: 3
  1 (nil): nil
  2 (string): exit
  3 (number): 22

os.execute: 	curl -k -f -L --user-agent "LuaRocks/3.2.1 macosx-x86_64 via curl" --connect-timeout 30 'http://luafr.org/moonrocks/manifest-5.1' > '/Users/idevz/.cache/luarocks/http___luafr.org_moonrocks_/manifest-5.1' 2> /dev/null
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

os.execute: 	curl -k -f -L --user-agent "LuaRocks/3.2.1 macosx-x86_64 via curl" --connect-timeout 30 'https://luarocks.org/manifest-5.1.zip' > '/Users/idevz/.cache/luarocks/https___luarocks.org/manifest-5.1.zip' 2> /dev/null
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

os.execute: 	unzip -n '-h'
UnZip 6.00 of 20 April 2009, by Info-ZIP.  Maintained by C. Spieler.  Send
bug reports using http://www.info-zip.org/zip-bug.html; see README for details.

Usage: unzip [-Z] [-opts[modifiers]] file[.zip] [list] [-x xlist] [-d exdir]
  Default action is to extract files in list, except those in xlist, to exdir;
  file[.zip] may be a wildcard.  -Z => ZipInfo mode ("unzip -Z" for usage).

  -p  extract files to pipe, no messages     -l  list files (short format)
  -f  freshen existing files, create none    -t  test compressed archive data
  -u  update files, create if necessary      -z  display archive comment only
  -v  list verbosely/show version info       -T  timestamp archive to latest
  -x  exclude files that follow (in xlist)   -d  extract files into exdir
modifiers:
  -n  never overwrite existing files         -q  quiet mode (-qq => quieter)
  -o  overwrite files WITHOUT prompting      -a  auto-convert any text files
  -j  junk paths (do not make directories)   -aa treat ALL files as text
  -C  match filenames case-insensitively     -L  make (some) names lowercase
  -X  restore UID/GID info                   -V  retain VMS version numbers
  -K  keep setuid/setgid/tacky permissions   -M  pipe through "more" pager
See "unzip -hh" or unzip.txt for more help.  Examples:
  unzip data1 -x joe   => extract all files except joe from zipfile data1.zip
  unzip -p foo | more  => send contents of foo.zip via pipe into program more
  unzip -fo foo ReadMe => quietly replace existing ReadMe if archive file newer
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

os.execute: 	unzip -n '/Users/idevz/.cache/luarocks/https___luarocks.org/manifest-5.1.zip'
Archive:  /Users/idevz/.cache/luarocks/https___luarocks.org/manifest-5.1.zip
  inflating: manifest-5.1
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0
Installing http://luafr.org/moonrocks/lua-protobuf-0.3.1-0.src.rock

os.execute: 	curl -k -f -L --user-agent "LuaRocks/3.2.1 macosx-x86_64 via curl" --connect-timeout 30 'http://luafr.org/moonrocks/lua-protobuf-0.3.1-0.src.rock' > '/private/var/folders/c0/7vhsg4s965g3w5wtzf7tkz_00000gn/T/luarocks_luarocks-rock-lua-protobuf-0.3.1-0-5090472/lua-protobuf-0.3.1-0.src.rock' 2> /dev/null
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

os.execute: 	unzip -n '/private/var/folders/c0/7vhsg4s965g3w5wtzf7tkz_00000gn/T/luarocks_luarocks-rock-lua-protobuf-0.3.1-0-5090472/lua-protobuf-0.3.1-0.src.rock'
Archive:  /private/var/folders/c0/7vhsg4s965g3w5wtzf7tkz_00000gn/T/luarocks_luarocks-rock-lua-protobuf-0.3.1-0-5090472/lua-protobuf-0.3.1-0.src.rock
  inflating: lua-protobuf-0.3.1-0.rockspec
   creating: lua-protobuf/
  inflating: lua-protobuf/LICENSE
  inflating: lua-protobuf/pb.c
  inflating: lua-protobuf/protoc.lua
  inflating: lua-protobuf/README.md
  inflating: lua-protobuf/test.lua
  inflating: lua-protobuf/pb.h
  inflating: lua-protobuf/luaunit.lua
  inflating: lua-protobuf/.travis.yml
   creating: lua-protobuf/rockspecs/
  inflating: lua-protobuf/rockspecs/lua-protobuf-scm-1.rockspec
  inflating: lua-protobuf/serpent.lua
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

Stopping after installing dependencies for lua-protobuf 0.3.1-0
```

Comment (membphis) [reply FAQ.md]: We need to mention the minimum version requirements.

If it's compatible with the lower version of Luarocks, it's even better.

**Comment by moonming** [comment]
I think this better in issue, not in the FAQ.

**Comment by idevz** [comment]
@moonming I think if this is an issue it should be pull to the luarocks repository? but many folks face this problem in our community. I think it's a base question for the new one to APISIX. So it should be a FAQ.

**Comment by membphis** [comment]
This problem cannot be solved by APISIX. So in the FAQ, I feel reasonable.

File: FAQ_CN.md [hunk 2]
```diff
@@ -56,3 +56,16 @@ APISIX 是当前性能最好的 API 网关，单核 QPS 达到 2.3 万，平均
 5. 高性能
 
 APISIX 需要一个配置中心，上面提到的很多功能是传统关系型数据库和KV数据库是无法提供的。与 etcd 同类软件还有 Consul、ZooKeeper等，更详细比较可以参考这里：[etcd why](https://github.com/etcd-io/etcd/blob/master/Documentation/learning/why.md#comparison-chart)，在将来也许会支持其他配置存储方案。
+
+## 为什么在用 Luarocks 安装 APISIX 依赖时会遇到超时，很慢或者不成功的情况？
+
+遇到 luarocks 慢的问题，有以下两种可能：
+
+1. luarocks 安装所使用的服务器被封
```
Comment (moonming) [inline FAQ_CN.md]: `被封` -> `不能访问`

File: FAQ_CN.md [hunk 3]
```diff
@@ -56,3 +56,16 @@ APISIX 是当前性能最好的 API 网关，单核 QPS 达到 2.3 万，平均
...
+2. 你所在的网络到 github 服务器之间有地方对 `git` 协议进行封锁
+
+针对第一个问题，你可以使用 https_proxy 或者使用 `--server` 选项来指定一个你可以访问或者访问更快的
+luarocks 服务。 运行 `luarocks config rocks_servers` 命令可以查看有哪些可用服务。
+
+如果翻墙都解决不了这个问题，那可以在安装的过程中添加 `--verbose` 查看具体是慢在什么地方。排除前面的
```
Comment (moonming) [inline FAQ_CN.md]: remove `如果翻墙都解决不了这个问题`

File: FAQ_CN.md [hunk 4]
```diff
@@ -56,3 +56,16 @@ APISIX 是当前性能最好的 API 网关，单核 QPS 达到 2.3 万，平均
 5. 高性能
 
 APISIX 需要一个配置中心，上面提到的很多功能是传统关系型数据库和KV数据库是无法提供的。与 etcd 同类软件还有 Consul、ZooKeeper等，更详细比较可以参考这里：[etcd why](https://github.com/etcd-io/etcd/blob/master/Documentation/learning/why.md#comparison-chart)，在将来也许会支持其他配置存储方案。
+
+## 为什么在用 Luarocks 安装 APISIX 依赖时会遇到超时，很慢或者不成功的情况？
```
Comment (moonming) [inline FAQ_CN.md]: you can also use docker to install APISIX.

Comment (idevz) [reply FAQ_CN.md]: yes, but docker isn't the only way, this choice is too much subjectivity

File: FAQ_CN.md (hunk 3 above)
Comment (idevz) [reply FAQ_CN.md]: I think change this to "如果使用代理仍然解决不了这个问题" will be more fluency

File: FAQ.md (hunk 1 above)
Comment (membphis) [reply FAQ.md]: any news?

**Comment by idevz** [comment]
@moonming any questions about this pr?

**Comment by moonming** [comment]
merged, thx

**Pull request merged**